src/screens/OtherProfile/hooks/useOtherProfileScreenHooks.tsx

```diff
@@ -0,0 +1,158 @@
+import {Context} from '../../../context';
+import {checkUserBlock, getOtherFeedsInProfile, getOtherProfile} from '../../../service/profile';
+import {setOtherProfileFeed} from '../../../context/actions/otherProfileFeed';
+
+const useOtherProfileScreenHooks = (targetUserProfileId: string, username: string) => {
+  /**
+   * State
+   */
+  const [isLoading, setIsLoading] = React.useState<boolean>(false);
+  const [isProfileFetching, setIsProfileFetching] = React.useState<boolean>(true);
+  const [isBlocked, setIsBlocked] = React.useState<boolean | undefined>(undefined);
+  const [isBlocking, setIsBlocking] = React.useState<boolean | undefined>(undefined);
+  const [otherProfileData, setOtherProfileData] = React.useState<any>({});
+  const [, setOtherProfileUserId] = React.useState<string>('');
+  const [postOffset, setPostOffset] = React.useState(0);
+
+  /**
+   * Other Profile Context
+   */
+  const [otherProfileFeeds, dispatchOtherProfile] = (React.useContext(Context) as any)
+    ?.otherProfileFeed;
+  const {feeds} = otherProfileFeeds;
+
+  /**
+   * Hook
+   */
+  const {mappingColorFeed} = useMappingPostColorHook();
+
+  const getCache = async () => {
+    const isBlockedFromCache = StorageUtils.blockedStatus.getForKey(targetUserProfileId) === 'true';
+    const isBlockingFromCache =
+      StorageUtils.blockingStatus.getForKey(targetUserProfileId) === 'true';
+
+    try {
+      const otherProfileDataCache =
+        StorageUtils.otherProfileData.getForKey(targetUserProfileId) || '{}';
+      setOtherProfileData(JSON.parse(otherProfileDataCache));
+      if (otherProfileDataCache !== '{}') {
+        setIsProfileFetching(false);
+      }
+    } catch (e) {
+      console.log('error parsing JSON other profile data', e);
+    }
+
+    try {
+      const feedsCache = StorageUtils.otherProfileFeed.getForKey(targetUserProfileId) || '{}';
+
+      setOtherProfileFeed(JSON.parse(feedsCache), dispatchOtherProfile);
+    } catch (e) {
+      console.log('error parsing JSON other profile feed', e);
+    }
+
+    setIsBlocked(isBlockedFromCache);
+    setIsBlocking(isBlockingFromCache);
+  };
+
+  const fetchFeeds = async (offset = 0) => {
+    setIsLoading(true);
+    try {
+      const result = await getOtherFeedsInProfile(targetUserProfileId, offset);
+      const {data: feedOtherProfile} = result;
+      const mapNewData = mappingColorFeed(feedOtherProfile);
+      if (offset === 0) {
+        setOtherProfileFeed(mapNewData, dispatchOtherProfile);
+        StorageUtils.otherProfileFeed.setForKey(targetUserProfileId, JSON.stringify(mapNewData));
+      } else {
+        const clonedFeeds = [...feeds, ...mapNewData];
+        setOtherProfileFeed(clonedFeeds, dispatchOtherProfile);
+        StorageUtils.otherProfileFeed.setForKey(targetUserProfileId, JSON.stringify(clonedFeeds));
+      }
+      setPostOffset(Number(result.offset));
+    } catch (e) {
+      console.log('error', e);
+    }
+    setIsLoading(false);
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `fetchFeeds` function, you're setting `isLoading` to true at the start of the function and to false at the end. If an error occurs during the execution of this function, `isLoading` will remain true indefinitely. To prevent this, consider moving `setIsLoading(false);` into a finally block.
```diff
- setIsLoading(true);
  try {
    //...
  } catch (e) {
    console.log('error', e);
  }
- setIsLoading(false);
+ finally {
+   setIsLoading(false);
+ }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
closed